### PR TITLE
pytest - MLIR binary ops

### PR DESCRIPTION
**Pull request by adam-smnk**

### CODE REVIEW

**Comment by adam-smnk** [comment]
@slyalin I took `test_add.py` as my test template. So far, when I run my test, I end up with:
`AssertionError: Operation mlir::binary_ops type doesn't exist in provided graph`
Not sure what I'm missing.

**Comment by rengolin** [comment]
Can you also add the test from #139, but change `input_shape` to `1024` and `output_shape` to `128` and changed all instances of `10` and `1` to `input_shape` and `output_shape` respectively?

**Comment by adam-smnk** [comment]
> Not sure what I'm missing.

Okay, I think I get it now. This `kind` looks for equivalent type of node in the input graph. For this test, I'm interested whether the model can run at all. So, I can probably set my return value `mlir::binary_ops` to `None` as it is checked before `compile_model` happens.

It would be nice if I could somehow run validation on MLIR but maybe we can leave that for later.

File: tests/layer_tests/pytorch_tests/test_mlir_binary.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    @pytest.mark.parametrize(("lhs_type", "rhs_type"), [[torch.float32, torch.float32]])
+    @pytest.mark.parametrize(("lhs_shape", "rhs_shape"), [([2, 3], [2, 3])])
+
+    def test_mlir_binary(self, ie_device, precision, ir_version, lhs_type, lhs_shape, rhs_type, rhs_shape):
+        self.lhs_type = lhs_type
+        self.lhs_shape = lhs_shape
+        self.rhs_type = rhs_type
+        self.rhs_shape = rhs_shape
+        # TODO: test with static shapes for XSMM acceleration
```
Comment (slyalin) [inline tests/layer_tests/pytorch_tests/test_mlir_binary.py:42]: @mvafin, do we have a way to reshape a model in this infrastructure?

Comment (adam-smnk) [reply tests/layer_tests/pytorch_tests/test_mlir_binary.py:42]: I was thinking to just set `dynamic_shapes=False` for that but right now I have no idea if I can validate what kind of MLIR we generate.

**Comment by adam-smnk** [comment]
Existing tests provide already decent coverage, for the rest benchmarks are currently enough.
Closing for now.

**Pull request closed without merging**